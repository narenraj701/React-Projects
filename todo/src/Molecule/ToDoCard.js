import React from 'react';
import Message from '../Atoms/ToDoMessage';
import Due from '../Atoms/Due';
import Mark from '../Atoms/MarkButton';
import Delete from '../Atoms/DeleteButton';

import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';

const useStyles = makeStyles({
    root: {
        position: 'relative',
        display: 'inline-block',
        margin: 10,
        width: 390,
        height: 240,
        padding: 5,
        boxShadow: '5px 5px 10px grey',
        textAlign: 'center',
    },
    position: {
        position: 'absolute',
        bottom: '10px',
        right: '60px'
    }
});

const ToDoCard = (props) => {
    const classes = useStyles();

    return (
        <Card className={classes.root}>
            <CardContent >
                <Message data-testid="message" message={props.message} finished={props.completed} />
                <Due due={props.due} />
            </CardContent>
            <CardActions className={classes.position}>
                <Mark disabled={props.completed} disableit={props.disableit} />
                <Delete delete={props.delete} />
            </CardActions>
        </Card>

    )
}

export default ToDoCard;