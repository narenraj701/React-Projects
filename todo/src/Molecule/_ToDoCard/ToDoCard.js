import React from 'react';
import Message from '../../Atoms/_ToDoMessage/ToDoMessage';
import Due from '../../Atoms/_Due/Due';
import Mark from '../../Atoms/_MarkButton/MarkButton';
import Delete from '../../Atoms/_DeleteButton/DeleteButton';

import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';

const useStyles = makeStyles({
    root: {
        position: 'relative',
        display: 'inline-block',
        margin: '10px 20px 10px 30px',
        width: 390,
        height: 240,
        padding: 5,
        boxShadow: '5px 5px 10px grey',
        textAlign: 'center',
    },
    position: {
        position: 'absolute',
        bottom: '10px',
        right: '60px'
    }
});

const ToDoCard = (props) => {
    const classes = useStyles();

    return (
        <Card className={classes.root}>
            <CardContent >
                <Message data-testid="message" message={props.message} finished={props.completed} />
                <Due due={props.due} />
            </CardContent>
            <CardActions className={classes.position}>
                <Mark disabled={props.completed} disableit={props.disableit} />
                <Delete delete={props.delete} />
            </CardActions>
        </Card>

    )
}

export default ToDoCard;