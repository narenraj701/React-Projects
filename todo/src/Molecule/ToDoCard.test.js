
import React from "react";
import { create } from "react-test-renderer";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import ToDoCard from './ToDoCard'


jest.mock('../Atoms/ToDoMessage',() => ({
    __esModule: true,
    default: () => <div data-testid="message">Message Component</div>
}))
jest.mock('../Atoms/Due',() => ({
    __esModule: true,
    default: () => <div data-testid="duemessage">DueMessage Component</div>
}))
jest.mock('../Atoms/MarkButton',() => ({
    __esModule: true,
    default: () => <div data-testid="markbutton">MarkButton Component</div>
}))
jest.mock('../Atoms/DeleteButton',() => ({
    __esModule: true,
    default: () => <div data-testid="deletebutton">deletebutton Component</div>
}))
test("Match the snapshot - Due message",() => {
    const input=create(<ToDoCard />);
    expect(input.toJSON()).toMatchSnapshot();
})
test("Check if it contains a Message Component",()=>{
    const {getByTestId}=render(<ToDoCard />);
    const messageElement=getByTestId('message');
    expect(messageElement).toBeInTheDocument();
})
test("check if it contains Due message ",()=>{
    const {getByTestId}=render(<ToDoCard />);
    const dueMessageElement=getByTestId('duemessage');
    expect(dueMessageElement).toBeInTheDocument();
})
test("check if it contains MarkButton ",()=>{
    const {getByTestId}=render(<ToDoCard />);
    const markButtonElement=getByTestId('markbutton');
    expect(markButtonElement).toBeInTheDocument();
})
test("check if it contains DeleteButton ",()=>{
    const {getByTestId}=render(<ToDoCard />);
    const deleteButtonElement=getByTestId('markbutton');
    expect(deleteButtonElement).toBeInTheDocument();
})
