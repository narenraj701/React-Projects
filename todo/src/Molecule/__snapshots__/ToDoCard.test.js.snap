// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match the snapshot - Due message 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardContent-root"
  >
    <div
      data-testid="message"
    >
      Message Component
    </div>
    <div
      data-testid="duemessage"
    >
      DueMessage Component
    </div>
  </div>
  <div
    className="MuiCardActions-root makeStyles-position-2 MuiCardActions-spacing"
  >
    <div
      data-testid="markbutton"
    >
      MarkButton Component
    </div>
    <div
      data-testid="deletebutton"
    >
      deletebutton Component
    </div>
  </div>
</div>
`;
